src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -857,34 +962,72 @@ public unsafe void CopyTo(Array array, int index)
                 {
                     Vector128<byte> lowerShuffleMask = s_lowerShuffleMask_CopyToBoolArray;
                     Vector128<byte> upperShuffleMask = s_upperShuffleMask_CopyToBoolArray;
-                    Vector128<byte> bitMask = Vector128.Create(0x80402010_08040201).AsByte(); ;
                     Vector128<byte> ones = Vector128.Create((byte)1);
 
                     fixed (bool* destination = &boolArray[index])
                     {
-                        for (; (i + Vector128<byte>.Count * 2) <= m_length; i += Vector128<byte>.Count * 2)
+                        for (; (i + Vector128ByteCount * 2u) <= (uint)m_length; i += Vector128ByteCount * 2u)
                         {
-                            int bits = m_array[i / BitsPerInt32];
+                            int bits = m_array[i / (uint)BitsPerInt32];
                             Vector128<int> scalar = Vector128.CreateScalarUnsafe(bits);
 
                             Vector128<byte> shuffledLower = Ssse3.Shuffle(scalar.AsByte(), lowerShuffleMask);
-                            Vector128<byte> extractedLower = Sse2.And(shuffledLower, bitMask);
+                            Vector128<byte> extractedLower = Sse2.And(shuffledLower, s_bitMask128);
                             Vector128<byte> normalizedLower = Sse2.Min(extractedLower, ones);
                             Sse2.Store((byte*)destination + i, normalizedLower);
 
                             Vector128<byte> shuffledHigher = Ssse3.Shuffle(scalar.AsByte(), upperShuffleMask);
-                            Vector128<byte> extractedHigher = Sse2.And(shuffledHigher, bitMask);
+                            Vector128<byte> extractedHigher = Sse2.And(shuffledHigher, s_bitMask128);
                             Vector128<byte> normalizedHigher = Sse2.Min(extractedHigher, ones);
                             Sse2.Store((byte*)destination + i + Vector128<byte>.Count, normalizedHigher);
                         }
                     }
                 }
+                else if (AdvSimd.IsSupported)
+                {
+                    Vector128<byte> ones = Vector128.Create((byte)1);
+                    fixed (bool* destination = &boolArray[index])
+                    {
+                        for (; (i + Vector128ByteCount * 2u) <= (uint)m_length; i += Vector128ByteCount * 2u)
+                        {
+                            int bits = m_array[i / (uint)BitsPerInt32];
+                            // Same logic as SSSE3 path, except we do not have Shuffle instruction.
+                            // (TableVectorLookup could be an alternative - dotnet/runtime#1277)
```

Comment:
I'd like a tracking issue to be logged for this, so we don't forget to replace the implementation once TableVectorLookup is implemented.

Reply:
FWIW, this should be addressed by https://github.com/dotnet/runtime/pull/38780.